pandas/tests/series/methods/test_update.py

```diff
@@ -39,34 +39,35 @@ def test_update(self, using_copy_on_write):
         tm.assert_frame_equal(df, expected)
 
     @pytest.mark.parametrize(
-        "other, dtype, expected",
+        "other, dtype, expected, warn",
         [
             # other is int
-            ([61, 63], "int32", Series([10, 61, 12], dtype="int32")),
-            ([61, 63], "int64", Series([10, 61, 12])),
-            ([61, 63], float, Series([10.0, 61.0, 12.0])),
-            ([61, 63], object, Series([10, 61, 12], dtype=object)),
+            ([61, 63], "int32", Series([10, 61, 12], dtype="int32"), None),
+            ([61, 63], "int64", Series([10, 61, 12]), None),
+            ([61, 63], float, Series([10.0, 61.0, 12.0]), None),
+            ([61, 63], object, Series([10, 61, 12], dtype=object), None),
             # other is float, but can be cast to int
-            ([61.0, 63.0], "int32", Series([10, 61, 12], dtype="int32")),
-            ([61.0, 63.0], "int64", Series([10, 61, 12])),
-            ([61.0, 63.0], float, Series([10.0, 61.0, 12.0])),
-            ([61.0, 63.0], object, Series([10, 61.0, 12], dtype=object)),
+            ([61.0, 63.0], "int32", Series([10, 61, 12], dtype="int32"), None),
+            ([61.0, 63.0], "int64", Series([10, 61, 12]), None),
+            ([61.0, 63.0], float, Series([10.0, 61.0, 12.0]), None),
+            ([61.0, 63.0], object, Series([10, 61.0, 12], dtype=object), None),
             # others is float, cannot be cast to int
-            ([61.1, 63.1], "int32", Series([10.0, 61.1, 12.0])),
-            ([61.1, 63.1], "int64", Series([10.0, 61.1, 12.0])),
-            ([61.1, 63.1], float, Series([10.0, 61.1, 12.0])),
-            ([61.1, 63.1], object, Series([10, 61.1, 12], dtype=object)),
+            ([61.1, 63.1], "int32", Series([10.0, 61.1, 12.0]), FutureWarning),
+            ([61.1, 63.1], "int64", Series([10.0, 61.1, 12.0]), FutureWarning),
+            ([61.1, 63.1], float, Series([10.0, 61.1, 12.0]), None),
+            ([61.1, 63.1], object, Series([10, 61.1, 12], dtype=object), None),
             # other is object, cannot be cast
-            ([(61,), (63,)], "int32", Series([10, (61,), 12])),
-            ([(61,), (63,)], "int64", Series([10, (61,), 12])),
-            ([(61,), (63,)], float, Series([10.0, (61,), 12.0])),
-            ([(61,), (63,)], object, Series([10, (61,), 12])),
+            ([(61,), (63,)], "int32", Series([10, (61,), 12]), FutureWarning),
+            ([(61,), (63,)], "int64", Series([10, (61,), 12]), FutureWarning),
+            ([(61,), (63,)], float, Series([10.0, (61,), 12.0]), FutureWarning),
+            ([(61,), (63,)], object, Series([10, (61,), 12]), None),
```

Comment:
is there a nice way of defining `warn` inside the test function e.g.

```
warn = None
if some_intutiive_condition:
    warn = FutureWarning
```

Reply:
to be honest I generally prefer to avoid logic (like if-then) statements in tests https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

can do if you prefer, just explaining why I intentionally did it this way

Reply:
ill give that a read.  my thought is that it is often not obvious why some cases have FutureWarning which can be a real problem when a test can be wrong(ish) eg https://github.com/pandas-dev/pandas/pull/50626#discussion_r1199642076